### ISSUE TITLE
Missing parentheses in homing.c?

### ISSUE THREAD

**Issue by hansu** [LinuxCNC/linuxcnc#2339]
Compiling gives this warning:

```
emc/motion/homing.c:1345:18: warning: suggest parentheses around operand of ‘!’ or change ‘&’ to ‘&&’ or ‘!’ to ‘~’ [-Wparentheses]
             if ( ! H[joint_num].home_flags & HOME_ABSOLUTE_ENCODER) {
                  ^~~~~~~~~~~~~~~~~~~~~~~~~
```

@andypugh  I guess your intention with

```c
if ( ! H[joint_num].home_flags & HOME_ABSOLUTE_ENCODER) {
```
was
```c
if ( !(H[joint_num].home_flags & HOME_ABSOLUTE_ENCODER) ) {
```
not
```c
if ( (!H[joint_num].home_flags) & HOME_ABSOLUTE_ENCODER ) {
```
which is actually equal to the first one.

**Comment by andypugh** [comment]
Yes, as you say, it is meant to be ! (flags & mask)
Though, it works as bitwise ! of the flags and then and-ing the mask is equivalent to masking the flags and then inverting.

**Comment by andypugh** [comment]
Fixed via 26e2ce16cfe95e519460415b08303c448b2c6028

**Comment by hansu** [comment]
Thanks
